### DISCUSSION TITLE
Unable to access Modbus RTU on USR-M100-ARD (Modbus Bridge)

### DISCUSSION THREAD

**Discussion by patkepa** [eModbus/eModbus#362] [Q&A] [answered]
Hello,

I'm trying to access Modbus RTU on USR-M100-ARD which is a a [IOT Gateway device](https://www.pusr.com/products/arduino-open-source-iot-gateway.html) which is based on ESP32-WROVER-E.

I'm having an issue when trying to access Modbus RTU over my device's pins:
![CleanShot 2024-07-02 at 11 18 26](https://github.com/eModbus/eModbus/assets/61762509/8ed8616e-91ba-4829-a209-487171560c00)

Where when trying to connect after setting them up, it displays this error:
![CleanShot 2024-07-02 at 11 20 19](https://github.com/eModbus/eModbus/assets/61762509/a3309064-1c81-4339-95a8-addc06de48e5)
I've checked for hardware issues, when connecting external Modbus analyser it works without issues.

Do you have any suggestions for making this work?

This is the simplified version of code I'm using.

  ```c
  RTUutils::prepareHardwareSerial(MODBUS_SERIAL);
  MODBUS_SERIAL.begin(config.getModbusBaudRate(), SERIAL_8N1,  GPIO_NUM_33, GPIO_NUM_32);

  MBclient = new ModbusClientRTU(GPIO_NUM_12); 
  MBclient->setTimeout(config.getRtuTimeout());
  MBclient->begin(MODBUS_SERIAL, 1);

  for (uint8_t i = 1; i <= MODBUS_MAX_ADDRESS; i++)
  {
    MBbridge.attachServer(i, i, ANY_FUNCTION_CODE, MBclient);
  }

  MBbridge.start(config.getTcpPort(), 10, config.getTcpTimeout());

```

Thanks for help!

**Comment by Miq1** [comment]
Interesting device! Unfortunately I currently have no access to my PC, only some mobile devices, so I am unable to open the rar-type documentation files.
I suppose you took the GPIO numbers for the serial port from those docs?
The baud rates of all servers connected to the RTU bus have been checked to be matching?
E0 TIMEOUT errors are common if there are connection problems or different communication settings. So I would advise to double-check these in the first row.

**Comment by patkepa** [comment, marked as answer]
Okay, thanks for a very quick reply!
I resolved the issue, it was actually outside of the scope of library, which works excellent now that I am testing it 😀

I've had issue with external device that was serving us Modbus RTU from Modbus TCP host outside of our network - our client's IT division blocked the connection by a mistake, and the _PUSR Serial Gateway_ that we're using, instead of providing an error, acted like it wasn't receiving anything on the serial. 

Cheers and have a good day!

**Reply by Miq1** [reply]
👍🏻
